### DISCUSSION TITLE
CmdFlags asynchronous

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#99] [Q&A]
Is it possible to create a function in asynchronous (non-UI blocking) mode? Is it necessary to create a new thread in the function?

**Comment by CEXT-Dan** [comment]
I’m not sure in what context, from a command perspective that would be  SESSION (I think), I.e. 

```Python
#session 
def PyRxCmd_session(CmdFlags = Ap.CmdFlags.SESSION):
    try:
        print(CmdFlags)
    except Exception as err:
        print(err)
```

**Comment by CEXT-Dan** [comment]
if you’re trying to call back into AutoCAD from a palette, or modeless dialog. You can use
DocManager.beginExecuteInApplicationContext
DocManager.beginExecuteInCommandContext

They don’t work on all platforms though, I will make a sample. I think these functions add the command to a queue

**Reply by CEXT-Dan** [reply]
added sample 0f603c4e7809f4f01e1ec7a8704db272452cdde3
